### ISSUE TITLE
Upgrade from Craft 2?

### ISSUE THREAD

**Issue by snowden0** [nfourtythree/craft3-entriessubset#10]
I'm updating a site from Craft 2 to 3. The old site used EntriesSubset, and I have installed the Craft 3 version on the new site. However, I'm still seeing notices saying ```Unable to find component class 'EntriesSubset'.```

Is it possible to retrieve the Craft 2 entry data and have the Craft 3 version of EntriesSubset read it?

**Comment by nfourtythree** [comment]
🤔 hmm, this is an interesting one. My initial reaction is that there is some data in the cache or DB that is not pointing to the correct class.

Would be interesting to see the difference in the data stored from Craft 2 to how it is in 3. 

If that is something you can access quickly please drop me some notes in here/screenshots. Otherwise, I will need to set aside some time to have a look into it, as I haven't written a direct migration from 2 -> 3 for this plugin at the moment

**Comment by snowden0** [comment]
I can’t seem to figure out how the v2 version stores content. I’ve looked at an entry in the v2 db that has an EntriesSubset field populated with content, but the craft_content table doesn’t have columns for that data. I’ve also poked around in craft_elements, craft_entries, and the various matrix tables and don’t see the content there either.

In the v2 db, I do see an EntriesSubset row in craft_plugins:
- id: 7
- class: EntriesSubset
- version: 0.5.4
- schemaVersion: 0.5.0
- enabled: 1

In the v3 db, the `class` column has been changed to `handle`. And there are now two rows for the plugin- one for the legacy install from v2 and one for the new install of the v3 plugin.

v3 db, legacy plugin row:
- id: 7,
- handle: entries-subset
- version: 0.5.4
- schemaVersion: 0.5.0

v3 db, new plugin row:
- id: 11
- handle: entriessubset
- version: 1.2.1
- schemaVersion: 1.1.0

Is “entriessubset” the intended handle for the new plugin? Maybe there's a conflict between the two?


- - -






On Jun 5, 2019, at 10:29 AM, Nathaniel Hammond <notifications@github.com<mailto:notifications@github.com>> wrote:


🤔 hmm, this is an interesting one. My initial reaction is that there is some data in the cache or DB that is not pointing to the correct class.

Would be interesting to see the difference in the data stored from Craft 2 to how it is in 3.

If that is something you can access quickly please drop me some notes in here/screenshots. Otherwise, I will need to set aside some time to have a look into it, as I haven't written a direct migration from 2 -> 3 for this plugin at the moment

**Comment by nfourtythree** [comment]
Yes `entriessubset` is the correct handle for v3. 

In the `craft_fields` table what is in the `type` column for your EntriesSubset fields

**Comment by snowden0** [comment]
In the v3 db, the `craft_fields` > `type` column contains “EntriesSubset”.


- - -






On Jun 5, 2019, at 11:18 AM, Nathaniel Hammond <notifications@github.com<mailto:notifications@github.com>> wrote:


Yes entriessubset is the correct handle for v3.

In the craft_fields table what is in the type column for your EntriesSubset fields

**Comment by nfourtythree** [comment]
It should contain `nfourtythree\entriessubset\fields\EntriesSubsetField` for example here is a row from my test environment:
 ![](http://x.n43.me/e4d7cc8d0dcd/Image%202019-06-05%20at%204.35.28%20pm.png)

**Comment by snowden0** [comment]
Ok I updated all of them but am now seeing an error on entries that use that field that says “Unable to find component class ‘nfourtythreeentriessubsetfieldsEntriesSubsetField’

[cid:46FE9415-583F-4E53-BBB8-3A21D9348053]


- - -






On Jun 5, 2019, at 11:35 AM, Nathaniel Hammond <notifications@github.com<mailto:notifications@github.com>> wrote:


It should contain nfourtythree\entriessubset\fields\EntriesSubsetField for example here is a row from my test environment:
[https://camo.githubusercontent.com/3a405055593886e81af1ccdf7d3db03663b0b2f9/687474703a2f2f782e6e34332e6d652f6534643763633864306463642f496d616765253230323031392d30362d30352532306174253230342e33352e3238253230706d2e706e67]<https://camo.githubusercontent.com/3a405055593886e81af1ccdf7d3db03663b0b2f9/687474703a2f2f782e6e34332e6d652f6534643763633864306463642f496d616765253230323031392d30362d30352532306174253230342e33352e3238253230706d2e706e67>

**Comment by snowden0** [comment]
Aha! I didn’t escape the slashes. I ran the update again and now everything seems to be working. Thanks!


- - -






On Jun 5, 2019, at 1:42 PM, John Manning <jmanning@wrayward.com<mailto:jmanning@wrayward.com>> wrote:

Ok I updated all of them but am now seeing an error on entries that use that field that says “Unable to find component class ‘nfourtythreeentriessubsetfieldsEntriesSubsetField’

<PastedGraphic-2.png>


- - -






On Jun 5, 2019, at 11:35 AM, Nathaniel Hammond <notifications@github.com<mailto:notifications@github.com>> wrote:


It should contain nfourtythree\entriessubset\fields\EntriesSubsetField for example here is a row from my test environment:
[https://camo.githubusercontent.com/3a405055593886e81af1ccdf7d3db03663b0b2f9/687474703a2f2f782e6e34332e6d652f6534643763633864306463642f496d616765253230323031392d30362d30352532306174253230342e33352e3238253230706d2e706e67]<https://camo.githubusercontent.com/3a405055593886e81af1ccdf7d3db03663b0b2f9/687474703a2f2f782e6e34332e6d652f6534643763633864306463642f496d616765253230323031392d30362d30352532306174253230342e33352e3238253230706d2e706e67>
